src/components/cc-zone-picker/cc-zone-picker.js

```diff
@@ -0,0 +1,174 @@
+import { css, html } from 'lit';
+import { CcFormControlElement } from '../../lib/form/cc-form-control-element.abstract.js';
+import { accessibilityStyles } from '../../styles/accessibility.js';
+import { i18n } from '../../translations/translation.js';
+import '../cc-badge/cc-badge.js';
+import '../cc-icon/cc-icon.js';
+import '../cc-img/cc-img.js';
+import '../cc-notice/cc-notice.js';
+import '../cc-zone-card/cc-zone-card.js';
+
+import { iconRemixEarthLine as zoneIcon } from '../../assets/cc-remix.icons.js';
+import { isStringEmpty } from '../../lib/utils.js';
+
+/**
+ * @typedef {import('./cc-zone-picker.types.js').ZoneItem} ZoneItem
+ * @typedef {import('./cc-zone-picker.types.js').ZoneSection} ZoneSection
+ * @typedef {import('../../lib/events.types.js').EventWithTarget<HTMLInputElement>} HTMLInputElementEvent
+ */
+
+/**
+ * A component that allows you to select a zone from a list of zones sections.
+ *
+ * @cssdisplay block
+ */
+export class CcZonePicker extends CcFormControlElement {
+  static get properties() {
+    return {
+      ...super.properties,
+      value: { type: String },
+      zonesSection: { type: Array },
+    };
+  }
+
+  constructor() {
+    super();
+
+    /** @type {Array<ZoneSection>} array of zones section */
+    this.zonesSection = [];
+
+    /** @type {string} current selected zone code */
+    this.value = null;
+  }
+
+  /**
+   * @param {HTMLInputElementEvent} e
+   */
+  _onZoneSelect(e) {
+    this.value = e.target.value;
+  }
+
+  render() {
+    return html`
+      <form>
+        <fieldset @change="${this._onZoneSelect}">
+          <legend>
+            <cc-icon class="zone-legend-icon" .icon="${zoneIcon}" size="lg"></cc-icon>
+            <span class="zone-legend-text"> ${i18n('cc-zone-picker.legend')} </span>
+          </legend>
+          <div class="zone-section">
+            ${this.zonesSection.map(
+              (zoneSection) => html`
+                ${!isStringEmpty(zoneSection.title)
+                  ? html` <div class="zone-section-title">${zoneSection.title}</div> `
+                  : ''}
+                <div class="form-controls">
+                  ${zoneSection.zones.map((zone) => this._renderZoneCard(zone, this.value))}
+                </div>
+              `,
+            )}
+          </div>
+        </fieldset>
+      </form>
+    `;
+  }
+
+  /**
+   * @param {ZoneItem} zone
+   * @param {string} selectedZoneCode
+   */
+  _renderZoneCard(zone, selectedZoneCode) {
+    const isSelected = zone.code === selectedZoneCode;
```

Comment:
suggestion: we don't really need the selected zone code so we could reduce the scope of the params

```js
/**
   * @param {ZoneItem} zone
   * @param {boolean} isCelected
   */
  _renderZoneCard(zone, isSelected) {
```

And use it like this:

```
${zoneSection.zones.map((zone) => this._renderZoneCard(zone, zone.code === this.value))}
```